ivadomed/object_detection/utils.py

```diff
@@ -0,0 +1,268 @@
+        size = [h_max - h_min, w_max - w_min, d_max - d_min]
+
+        if length is not None and stride is not None:
+            for idx, dim in enumerate(size):
+                if dim < length[idx]:
+                    size[idx] = length[idx]
+            # Adjust size according to stride to avoid dimension mismatch
+            size = resize_to_multiple(size, stride, length)
+        index_shape.append(tuple(size))
+        transform_obj = imed_transforms.BoundingBoxCrop(size=size)
+        transforms.transform[img_type].transforms.insert(resample_idx[i] + 1, transform_obj)
+
+    for metadata in seg_pair['input_metadata']:
+        assert len(set(index_shape)) == 1
+        metadata['index_shape'] = index_shape[0]
+    return transforms
+
+
+def adjust_undo_transforms(transforms, seg_pair, index=0):
+    """
+    This function adapts the undo transforms by adding the BoundingBoxCrop to undo transform according the specific
+    parameters of an image.
+    :param transforms: transforms (Compose object)
+    :param seg_pair: segmentation pair (input, gt and metadata)
+    :param index: batch index of the seg_pair
+    :return:
+    """
+    for img_type in transforms.transform:
+        resample_idx = -1
+        for idx, transfo in enumerate(transforms.transform[img_type].transforms):
+            if "Resample" == transfo.__class__.__name__:
+                resample_idx = idx
+            if "BoundingBoxCrop" == transfo.__class__.__name__:
+                transforms.transform[img_type].transforms.pop(idx)
+        if "bounding_box" in seg_pair['input_metadata'][index][0]:
+            size = list(seg_pair['input_metadata'][index][0]['index_shape'])
+            transform_obj = imed_transforms.BoundingBoxCrop(size=size)
+            transforms.transform[img_type].transforms.insert(resample_idx + 1, transform_obj)
+
+
+def load_bounding_boxes(object_detection_params, subjects, slice_axis, constrast_lst):
+    """
+    Verifies if bounding_box.json exists in the log directory, if so loads the data, else generates the file if a valid
+    detection model path exists.
+    :param object_detection_params:
+    :param subjects:
+    :param slice_axis:
+    :param constrast_lst:
+    :return:
+    """
+    # Load or generate bounding boxes and save them in json file
+    bounding_box_dict = {}
+    if object_detection_params is None or object_detection_params['object_detection_path'] is None:
+        return bounding_box_dict
+    bounding_box_path = os.path.join(object_detection_params['log_directory'], 'bounding_boxes.json')
+    if os.path.exists(bounding_box_path):
+        with open(bounding_box_path, 'r') as fp:
+            bounding_box_dict = json.load(fp)
+    elif object_detection_params['object_detection_path'] is not None and \
+            os.path.exists(object_detection_params['object_detection_path']):
+        print("Generating bounding boxes...")
```

Comment:
we should avoid using `print()` in API functions, but i guess this is out-of-scope https://github.com/neuropoly/ivadomed/issues/10